extension-base/src/main/java/com/azure/autorest/extension/base/util/FileUtils.java

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.autorest.extension.base.util;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/**
+ * Utility class for file operations.
+ */
+public final class FileUtils {
+    private FileUtils() {
+    }
+
+    /**
+     * Creates a temporary directory.
+     * <p>
+     * If the environment setting {@code codegen.java.temp.directory} is set, the directory will be created under the
+     * specified path. Otherwise, the directory will be created under the system default temporary directory.
+     * <p>
+     * {@link System#getProperty(String)} is checked before {@link System#getenv(String)}.
+     * <p>
+     * If {@code codegen.java.temp.directory} is set to a non-existent path, the directory will be created under the
+     * system default temporary directory.
+     *
+     * @param prefix The prefix string to be used in generating the directory's name; may be {@code null}.
+     * @return The path to the newly created directory.
+     * @throws IOException If an I/O error occurs.
+     */
+    public static Path createTempDirectory(String prefix) throws IOException {
+        String tempDirectory = System.getProperty("codegen.java.temp.directory");
+        if (tempDirectory == null) {
+            tempDirectory = System.getenv("codegen.java.temp.directory");
+        }
+
+        if (tempDirectory != null) {
+            Path tempDirectoryPath = Paths.get(tempDirectory);
+            if (Files.exists(tempDirectoryPath)) {
+                return Files.createTempDirectory(tempDirectoryPath, prefix);
+            }
```

Comment:
Since this package already depends on azure-core, we can consider using `Configuration` to get system property or env var with a single method call.

Reply:
I initially left this out as I don't know how tightly we want to couple code generation to azure-core as it becomes a bit of a circular dependency at times.

Reply:
Autorest/codegen will always take a dependency on azure-core package as we need to use the types defined in azure-core. So, this dependency is expected but the opposite direction for core to depend on autorest shouldn't be allowed and we should not have circular dependency.